packages/pyright-internal/src/common/serviceProviderExtensions.ts

```diff
@@ -77,6 +77,10 @@ ServiceProvider.prototype.console = function () {
     return this.get(ServiceKeys.console);
 };
 ServiceProvider.prototype.partialStubs = function () {
+    const result = this.tryGet(ServiceKeys.partialStubs);
+    if (!result) {
+        this.add(ServiceKeys.partialStubs, new PartialStubService(this.fs()));
```

Comment:
This looks good. I'm curious though why in other functions (ex. `docStringService` below we create a new instance if needed, but don't save it in the service provider.